components/modules/mqtt.c

```diff
@@ -634,13 +652,20 @@ static int mqtt_new( lua_State* L )
 	clientid = luaL_checkstring( L, 1 );
 	NODE_DBG("MQTT client id %s\n", clientid);
 
-	esp_mqtt_client_config_t * mqtt_cfg = (esp_mqtt_client_config_t *) malloc(sizeof(esp_mqtt_client_config_t));
-	memset(mqtt_cfg, 0, sizeof(esp_mqtt_client_config_t));
+	esp_mqtt_lua_client_config_t * mqtt_cfg = (esp_mqtt_lua_client_config_t *) malloc(sizeof(esp_mqtt_lua_client_config_t));
+	memset(mqtt_cfg, 0, sizeof(esp_mqtt_lua_client_config_t));
+	mqtt_cfg->config.host = mqtt_cfg->host;
```

Comment:
When compiling locally, gcc complains:
> components/modules/mqtt.c:657:24: error: assignment to expression with array type
>  mqtt_cfg->config.host = mqtt_cfg->host;

No clue why the [travis build](https://travis-ci.org/nodemcu/nodemcu-firmware/builds/438341577?utm_source=github_status&utm_medium=notification) passes without issues.